Frontend/src/components/cluster-details/user-registration-form.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import type { CascaderProps } from 'antd';
+import {
+  Button,
+  Form,
+  Input,
+  message,
+  Select,
+} from 'antd';
+import { USER_ROLES } from '@/data/user-data';
+import { registerUser } from '@/services/auth-service';
+
+const { Option } = Select;
+
+interface DataNodeType {
+  value: string;
+  label: string;
+  children?: DataNodeType[];
+}
+
+const formItemLayout = {
+  labelCol: {
+    xs: { span: 24 },
+    sm: { span: 8 },
+  },
+  wrapperCol: {
+    xs: { span: 24 },
+    sm: { span: 16 },
+  },
+};
+
+const tailFormItemLayout = {
+  wrapperCol: {
+    xs: {
+      span: 24,
+      offset: 0,
+    },
+    sm: {
+      span: 16,
+      offset: 8,
+    },
+  },
+};
+
+type props = {
+  onSuccess: () => void;
+}
+
+const UserRegistrationForm = ({onSuccess}: props) => {
+  const [form] = Form.useForm();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const onFinish = async (values: any) => {
+    setLoading(true);
+    try{
+      const res = await registerUser(values.username, values.email, values.password, values.role);
+      message.loading("Creating profile", 2);
+      onSuccess();
+    }catch(err){
+      message.error("Failed to create profile");
+    }
+    setLoading(false);
+  };
+
+  return (
+    <Form
+      {...formItemLayout}
+      form={form}
+      name="register"
+      onFinish={onFinish}
+      style={{ maxWidth: 600 }}
+      scrollToFirstError
+    >
+      <Form.Item
+        name="username"
+        label="Username"
+        tooltip="What do you want others to call you?"
```

Comment:
Can we have better tool tip text than this?

Reply:
i added this "Enter your username. Keep it simple and recognizable." is it better??